src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -2670,6 +2673,13 @@ static bool isAcceptableRefMismatch(RefKind refKind, bool isInterpolatedStringHa
             }
         }
 
+        private static RefKind GetParameterBetternessRefKind<TMember>(ParameterSymbol parameter, TMember member) where TMember : Symbol
+        {
+            // For static extension members, the ref kind of the extension parameter shouldn't affect betterness.
+            bool isExtensionParameterOfStaticExtensionMember = parameter is { ContainingSymbol: TypeSymbol { IsExtension: true } } && member.IsStatic;
```

Comment:
nit: it might also be good to check `parameter == parameter.ContainingSymbol.ExtensionParameter`.